{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Set Up\n",
    "- access to HPC system of CAU\n",
    "- set up conda environments for the bioinformatic pipelines\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Installation of SSH client\n",
    "## Log-in steps\n",
    "- Bitvise SSH Client\n",
    "- Host: nesh-login.rz.uni-kiel.de\n",
    "- Port: 22\n",
    "- Username: smomw681\n",
    "- Initial method: password\n",
    "- Enter password and log in \n",
    "-> access to SFTP and terminal directly connected the HPC system\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "plaintext"
    }
   },
   "outputs": [],
   "source": [
    "# Alternatively:\n",
    "ssh -X smomw681@nesh-login.rz.uni-kiel.de\n",
    "login "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If there is any problem:  hpcsupport@rz.uni-kiel.de\n",
    "\n",
    "The oparating system is Rocky Linux 8.6 with default system complier gcc v8.5.0\n",
    "\n",
    "After Acoount_login: \n",
    "\n",
    "$WORK quota set to 10 TB (hard), 9 TB (soft) \n",
    "\n",
    "Absolute path to my storage: /gxfs_work/geomar/smomw681\n",
    "Relative path: $WORK\n",
    "\n",
    "\n",
    "Before doing anything else, load gcc complier and conda and/or python modules:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "plaintext"
    }
   },
   "outputs": [],
   "source": [
    "cd $WORK\n",
    "module load gcc12-env/12.3.0\n",
    "module load miniconda3/24.11.1\n",
    "\n",
    "# show all available software modules\n",
    "module all\n",
    "\n",
    "# show all currently available modules in the currently active environment\n",
    "module avail\n",
    "\n",
    "# some possibly useful modules\n",
    "module load cluster-env/default\n",
    "module load matlab-geomar/R2023a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "plaintext"
    }
   },
   "outputs": [],
   "source": [
    "## How to use python env\n",
    "module load gcc12-env/12.3.0\n",
    "module load python/3.11.5\n",
    "# create a python environment\n",
    "mkdir $HOME/my_python_env\n",
    "python -m venv $HOME/my_python_env/my_env\n",
    "# install package into env\n",
    "source $HOME/my_python_env/my_env/bin/activate\n",
    "module load gcc/12.3.0\n",
    "pip install ...\n",
    "deactivate\n",
    "# Use the installed package \n",
    "module load gcc12-env/12.3.0\n",
    "source $HOME/my_python_env/my_env/bin/activate\n",
    "module load python/3.11.5\n",
    "...\n",
    "deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Access control \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "plaintext"
    }
   },
   "outputs": [],
   "source": [
    "ls -l /gxfs_work/<account to access>/\n",
    "ls -l /gxfs_work/geomar | grep <account to access>\n",
    "getfacl /gxfs_work/geomar/<account to access>/\n",
    "\n",
    "setfacl -m u:<me> /gxfs_work/geomar/<account to access>\n",
    "getfacl /gxfs_work/geomar/<account to access>/\n",
    "\n",
    "# assess size of folders\n",
    "du -sh dir_path\n",
    "# assess size of files\n",
    "ls -lh file_path"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Set up of required environment\n",
    "The conda package is already installed in the working directory of the smomw681. \n",
    "Conda environments for each of the pipeline are required, containing following bioconda, other packages and modules:\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "plaintext"
    }
   },
   "outputs": [],
   "source": [
    "# For each and every enviroments: \n",
    "    module load gcc12-env/12.3.0\n",
    "    module load miniconda3/24.11.1\n",
    "    module load parallel/20230822\n",
    "    module load boost/1.83.0\n",
    "    module load gcc/12.3.0\n",
    "    module load gsl/2.7.1\n",
    "    module load nano/7.2\n",
    "\n",
    "# list of my conda environments:\n",
    "conda env list\n",
    "\n",
    "# list of packages in the environment\n",
    "conda list -n myenv\n",
    "\n",
    "#update every packages in the environment\n",
    "conda update --all -y\n",
    "\n",
    "# Preprocessing: \n",
    "    conda create --name FastqDump -y\n",
    "    conda activate FastqDump\n",
    "    # basic modules loaded\n",
    "    conda install bioconda::sra-tools=3.2.0 -y\n",
    "    conda install bioconda::parallel-fastq-dump=0.6.7 -y \n",
    "    conda install conda-forge::parallel=20241222 -y\n",
    "    conda install conda-forge::ncbi-datasets-cli=16.40.1 -y\n",
    "    conda install bioconda::ncbi-genome-download=0.3.3 -y\n",
    "\n",
    "# Assembly\n",
    "    conda create --name Assembly -y \n",
    "    conda activate Assembly    \n",
    "    conda install conda-forge::libgcc-ng=14.2.0     # for the current version of spades and bbmap\n",
    "    conda install bioconda::spades=4.0.0 -y   #or v4.1.0\n",
    "    conda install bioconda::trimmomatic=0.39 -y  \n",
    "    conda install bioconda::bbmap # v39.18        \n",
    "    conda install conda-forge::pigz=2.8 -y\n",
    "    \n",
    "# PacBio_Assembly\n",
    "    conda create -n PacBio_Assembly\n",
    "    conda activate PacBio_Assembly\n",
    "    conda install bioconda::wtdbg=2.5 -y\n",
    "    conda install bioconda::quast=5.3.0 -y\n",
    "    conda install bioconda::flye=2.9.5 -y\n",
    "    conda install bioconda::canu=2.3 -y\n",
    "    conda install bioconda::unicycler=0.5.1 -y\n",
    "    conda install bioconda::fastqc=0.12.1 -y \n",
    "    conda install bioconda::multiqc=1.27.1 -y\n",
    "    conda install conda-forge::parallel=20241222 -y\n",
    "    conda install bioconda::trycycler=0.5.5 -y\n",
    "    conda install bioconda::filtlong=0.2.1 -y\n",
    "    conda install bioconda::bandage=0.8.1 -y\n",
    "    # due to mash dependencies error, trycycler package is installed in separate env\n",
    "\n",
    "# Trycycler (raven, flye, hifiasm)\n",
    "    conda create -c bioconda -c conda-forge -n trycycler trycycler # v0.5.5\n",
    "    conda install bioconda::raven-assembler=1.8.3\n",
    "    conda install -c bioconda pbmm2 pb-assembly pbbam # v1.17.0, v0.0.7, v2.4.0\n",
    "    conda install bioconda::any2fasta=0.4.2 \n",
    "    conda install bioconda::hifiasm=0.25.0\n",
    "    conda install bioconda::hifiasm_meta=v0.3.2\n",
    "    conda install bioconda::flye=2.9.5\n",
    "    conda install bioconda::minipolish=0.1.3\n",
    "\n",
    "# MAG prep and after\n",
    "    conda create -n MAG\n",
    "    conda activate MAG\n",
    "    #conda install bioconda::deepmicroclass=1.0.3 -y \n",
    "    conda install python=3.12.9 -y\n",
    "    pip install DeepMicroClass # without pytorch: use CPU only\n",
    "    pip install requests #v2.32.3\n",
    "    conda install bioconda::checkm-genome -y #=1.2.3\n",
    "    conda install bioconda::prodigal -y #=2.6.3\n",
    "    conda install bioconda::pprodigal -y #=1.0.1\n",
    "    conda install bioconda::gtdbtk -y #=2.4.0\n",
    "    # db for gtdbtk also downloaded: run script 3_0 \n",
    "    conda install bioconda/label/cf201901::mash # v2.3\n",
    "\n",
    "# MAG_construction and check\n",
    "    conda create -n METABAT2\n",
    "    conda activate METABAT2\n",
    "    conda install bioconda::metabat2=2.17\n",
    "    # metabat2 must be installed anew according to the description on the official github site due to negative coverage problem\n",
    "    # https://bitbucket.org/berkeleylab/metabat/src/master/INSTALL.md\n",
    "    module load gcc12-env/12.3.0\n",
    "    module load gcc/12.3.0\n",
    "    module load boost/1.83.0\n",
    "    module load cmake/3.27.4\n",
    "    conda install bioconda::checkm2=1.1.0\n",
    "    # checkm2 database in /gxfs_work/geomar/smomw681/.conda/envs/METABAT2/checkm_data/\n",
    "    # or download it to a custom directory and set as environmental variable\n",
    "        checkm2 database --download --path  /gxfs_work/geomar/smomw681/DATABASES/CheckM_db/\n",
    "        export CHECKM2DB=\"/gxfs_work/geomar/smomw681/DATABASES/CheckM_db/CheckM2_database\"\n",
    "        conda env config vars set CHECKM2DB=\"/gxfs_work/geomar/smomw681/DATABASES/CheckM_db/CheckM2_database\"\n",
    "    conda install bioconda::coverm=0.7.0\n",
    "    conda install bioconda::drep # \n",
    "    # install dependencies of drep\n",
    "    conda install bioconda/label/cf201901::mash # v2.1\n",
    "    module load boost/1.83.0\n",
    "    module load gcc/12.3.0\n",
    "    module load gsl/2.7.1\n",
    "    conda install bioconda::centrifuge -y # v1.0.4.2\n",
    "    # conda install bioconda/label/cf201901::fastani -y # v1.1\n",
    "        # But couldn't solve the dependency problem of fastANI. conflicting with gsllib\n",
    "        # made a separate environment because of dependency problem\n",
    "    \n",
    "    \n",
    "\n",
    "# HiFi-MAG MAG_construction (might be better method next time)\n",
    "    conda create -n HiFi_MAG \n",
    "    conda activate HiFi_MAG\n",
    "    conda install bioconda::snakemake\n",
    "\n",
    "\n",
    "# dRep \n",
    "    conda create -n dRep\n",
    "    conda activate dRep\n",
    "    module load boost/1.83.0\n",
    "    module load gcc/12.3.0\n",
    "    module load gsl/2.7.1\n",
    "    conda install bioconda/label/cf201901::mash # v2.1\n",
    "    conda install bioconda::centrifuge -y # v1.0.4.2\n",
    "    conda install bioconda/label/cf201901::fastani -y # v1.1\n",
    "    conda install bioconda::checkm2 -y # v1.1.0\n",
    "    conda install bioconda::drep # v3.5.0\n",
    "    # it finally worked! Install the dependencies first and then the main package\n",
    "\n",
    "#GTDB-TK\n",
    "    conda create -n GTDBTK bioconda::gtdbtk #v2.4.0\n",
    "    conda activate GTDBTK\n",
    "    download-db.sh  # or alternatively copy it from the conda env \"MAG\n",
    "    cd /gxfs_work/geomar/smomw681/DATABASES/\n",
    "    mkdir GTDBTK_db\n",
    "    cp -R /gxfs_work/geomar/smomw681/.conda/envs/MAG/share/gtdbtk-2.4.0/db GTDBTK_db/\n",
    "    # conda env config vars set GTDBTK_DATA_PATH=\"/gxfs_work/geomar/smomw681/DATABASES/GTDBTK_db/db\"\n",
    "    conda install bioconda::mash=2.3 \n",
    "    # libgsl causes some dependency problem, so create symbolic link for libgsl.25.0\n",
    "    ln -s /gxfs_work/geomar/smomw681/.conda/envs/GTDBTK/lib/libgsl.so.27 /gxfs_work/geomar/smomw681/.conda/envs/GTDBTK/lib/libgsl.so.25\n",
    "    ls -l /gxfs_work/geomar/smomw681/.conda/envs/GTDBTK/lib/libgsl.so.25\n",
    "    # lrwxrwxrwx 1 smomw681 smomw 62 Apr  4 03:37 /gxfs_work/geomar/smomw681/.conda/envs/GTDBTK/lib/libgsl.so.25 -> /gxfs_work/geomar/smomw681/.conda/envs/GTDBTK/lib/libgsl.so.27\n",
    "\n",
    "\n",
    "# AntiSMASH\n",
    "    conda create --name AntiSMASH\n",
    "    conda activate AntiSMASH\n",
    "    conda install conda-forge::scikit-learn=1.6.1 -y\n",
    "    conda install conda-forge::icu=73.2 -y\n",
    "       # - current : 75.1 # for bioconda::meme\n",
    "    conda install python=3.10\n",
    "       # - current : 3.13.1\n",
    "    conda install bioconda::meme=5.8.5 -y\n",
    "    conda install hmmer2 hmmer diamond fasttree prodigal blast\n",
    "    conda install bioconda::antismash -y\n",
    "\n",
    "    # Previous error with installation of AntiSMASH: tt didn't work so started from the scratch\n",
    "    conda remove -n ENV_NAME --all\n",
    "    conda create --name AntiSMASH\n",
    "    conda activate AntiSMASH\n",
    "    #load modules\n",
    "    conda install bioconda::antismash -y \n",
    "    # error: Could not solve for environment specs\n",
    "The following packages are incompatible\n",
    "└─ antismash is not installable because there are no viable options\n",
    "   ├─ antismash [4.0.1|4.0.2|4.1.0] would require\n",
    "   │  └─ scikit-learn 0.18.2 , which does not exist (perhaps a missing channel);\n",
    "   ├─ antismash 4.2.0 would require\n",
    "   │  └─ scikit-learn 0.18.* , which does not exist (perhaps a missing channel);\n",
    "   ├─ antismash 4.2.0 would require\n",
    "   │  └─ icu >=64.2,<65.0a0 , which does not exist (perhaps a missing channel);\n",
    "   ├─ antismash [5.1.1|5.1.2|...|6.1.1] would require\n",
    "   │  └─ pyscss, which does not exist (perhaps a missing channel);\n",
    "   └─ antismash 7.1.0 would require\n",
    "      └─ libsass, which does not exist (perhaps a missing channel).    \n",
    "    conda install conda-forge::scikit-learn=1.6.1 -y \n",
    "    # but removed it because it isn't recognized\n",
    "    conda install conda-forge::libsass=0.22.0 -y\n",
    "    conda install conda-forge/label/python_rc::_python_rc -y\n",
    "    conda install python=3.10 -y\n",
    "    conda install bioconda::antismash=7.1.0 -y \n",
    "    download-antismash-databases\n",
    "    conda install hmmer2 hmmer diamond fasttree prodigal blast\n",
    "    # antismash my_input.gbk # for use\n",
    "    # install multismash for ARG statistics\n",
    "    git clone https://github.com/zreitz/multismash.git  /gxfs_work/geomar/smomw681/.conda/envs/AntiSMASH/bin/\n",
    "    cd  /gxfs_work/geomar/smomw681/.conda/envs/AntiSMASH/bin/multismash\n",
    "    pip install . --user\n",
    "\n",
    "# deepbgc \n",
    "    conda create -n DeepBGC python\n",
    "    conda install bioconda::hmmer\n",
    "    conda install bioconda::prodigal\n",
    "    pip install deepbgc # v0.1.31\n",
    "    deepbgc download    # download the database\n",
    "\n",
    "\n",
    "# Deep ARG\n",
    "    conda create --name DeepARG\n",
    "    conda activate DeepARG\n",
    "    conda install bioconda::deeparg=1.0.4 -y\n",
    "    conda install bioconda::diamond\n",
    "    pip install git+https://github.com/gaarangoa/deeparg.git\n",
    "    deeparg download_data -o \"/gxfs_work/geomar/smomw681/DATABASES/DeepARG\"\n",
    "    pip install deepbgc # alternatively with conda install bioconda::deepbgc=0.1.31, but warning on official site\n",
    "    deepbgc download\n",
    "\n",
    "\n",
    "\n",
    "# Check where a package/module is coming from:\n",
    "which\n",
    "\n",
    "# To exit from the module:\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "plaintext"
    }
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "language_info": {
   "name": "python"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
