{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "bc09eb9d",
   "metadata": {},
   "source": [
    "# Bioinformatic analysis for Nanopore 16S Amplicon Sequencing "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7fd0a05a",
   "metadata": {
    "vscode": {
     "languageId": "plaintext"
    }
   },
   "outputs": [],
   "source": [
    "# set up the conda environment named DORADO 1.3.0\n",
    "    module load gcc12-env/12.3.0\n",
    "    module load miniconda3/24.11.1\n",
    "    conda create -n DORADO_1.3.0\n",
    "    conda activate DORADO_1.3.0\n",
    "    cd /gxfs_work/geomar/smomw681/DORADO\n",
    "    wget -r \"https://cdn.oxfordnanoportal.com/software/analysis/dorado-1.3.0-linux-x64.tar.gz\"\n",
    "    tar -xzvf cdn.oxfordnanoportal.com/software/analysis/dorado-1.3.0-linux-x64.tar.gz\n",
    "    dorado-1.3.0-linux-x64/bin/dorado --version\n",
    "    echo 'export PATH=\"/gxfs_work/geomar/smomw681/DORADO/dorado-1.3.0-linux-x64/bin:$PATH\"' >> ~/.bashrc\n",
    "    source ~/.bashrc\n",
    "\n",
    "# Download the model required\n",
    "wget -O /gxfs_work/geomar/smomw681/DORADO/dna_r10.4.1_e8.2_400bps_hac@v5.2.0.zip -r \"https://cdn.oxfordnanoportal.com/software/analysis/dorado/dna_r10.4.1_e8.2_400bps_hac@v5.2.0.zip\" \n",
    "unzip /gxfs_work/geomar/smomw681/DORADO/dna_r10.4.1_e8.2_400bps_hac@v5.2.0.zip\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "621422ac",
   "metadata": {},
   "source": [
    "## Sample specification and experiment information\n",
    "- Sample specification: Sample name (containing Sampling date, type, location and spot), short version \n",
    "    - without year (2025) and location as P1(SPZO1), P2(SPZO2), P3(POST)\n",
    "- Experiment information: barcode and batch \n",
    "    - Additional experiments with the DNA sample or corresponding RNA sample \n",
    "        - 16S: 16S rRNA V3/V4 sequencing on Illumina platform\n",
    "        - metaWGS: metagenomic whole genome sequencing on DNB-Seq platform\n",
    "        - metaRNA: metatranscriptomic sequencing on DNB-Seq platform \n",
    "\n",
    "- Format\n",
    "    Barcode.    short sample name   Other experiments\n",
    "\n",
    "###\n",
    "### Sequencing Run 1\n",
    "in N16S_Run1_barcode_sample_table.txt\n",
    "1. 0506_P1_FW_2_1   16S\n",
    "2. 0526_P1_FW_2_1   16S\n",
    "3. 0612_P1_FW_2_1   16S\n",
    "4. 0626_P1_FW_2_1   16S\n",
    "5. 0710_P1_FW_2_1   16S\n",
    "6. 0724_P1_FW_2_1   16S\n",
    "7. 0804_P1_FW_2_1   16S\n",
    "8. 0820_P1_FW_2_1   16S\n",
    "\n",
    "9. 0506_sp2_1   16S, metaWGS, metaRNA\n",
    "10. 0526_sp2_2   16S, metaWGS, metaRNA\n",
    "11. 0612_sp2_2   16S, metaWGS, metaRNA\n",
    "12. 0626_sp2_1   16S, metaWGS, metaRNA\n",
    "13. 0710_sp2_2   16S, metaWGS, metaRNA\n",
    "14. 0724_sp2_1   16S, metaWGS, metaRNA\n",
    "15. 0804_sp2_2_2   16S, metaWGS, metaRNA\n",
    "16. 0820_sp2_2   16S, metaWGS, metaRNA\n",
    "\n",
    "17. 0506_sp3_2   16S, metaWGS, metaRNA\n",
    "18. 0526_sp3_2   16S, metaWGS, metaRNA\n",
    "19. 0612_sp3_2   16S, metaWGS, metaRNA\n",
    "20. 0626_sp3_2   16S, metaWGS, metaRNA\n",
    "21. 0710_sp3_3_2   16S, metaWGS, metaRNA\n",
    "22. 0724_sp3_3   16S, metaWGS, metaRNA\n",
    "23. 0804_sp3_3   16S, metaWGS, metaRNA\n",
    "24. 0820_sp3_1   16S, metaWGS, metaRNA\n",
    "\n",
    "### Sequencing Run 2\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "214bbac9",
   "metadata": {},
   "source": [
    "## Samp"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "name": "python"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
