{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "bc09eb9d",
   "metadata": {},
   "source": [
    "# Bioinformatic analysis for Nanopore 16S Amplicon Sequencing "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "afff1f45",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7fd0a05a",
   "metadata": {
    "vscode": {
     "languageId": "plaintext"
    }
   },
   "outputs": [],
   "source": [
    "# set up the conda environment named DORADO 1.3.0\n",
    "    module load gcc12-env/12.3.0\n",
    "    module load miniconda3/24.11.1\n",
    "    conda create -n DORADO_1.3.0\n",
    "    conda activate DORADO_1.3.0\n",
    "    cd /gxfs_work/geomar/smomw681/DORADO\n",
    "    wget -r \"https://cdn.oxfordnanoportal.com/software/analysis/dorado-1.3.0-linux-x64.tar.gz\"\n",
    "    tar -xzvf cdn.oxfordnanoportal.com/software/analysis/dorado-1.3.0-linux-x64.tar.gz\n",
    "    dorado-1.3.0-linux-x64/bin/dorado --version\n",
    "    echo 'export PATH=\"/gxfs_work/geomar/smomw681/DORADO/dorado-1.3.0-linux-x64/bin:$PATH\"' >> ~/.bashrc\n",
    "    source ~/.bashrc\n",
    "\n",
    "# Download the model required\n",
    "wget -O /gxfs_work/geomar/smomw681/DORADO/dna_r10.4.1_e8.2_400bps_hac@v5.2.0.zip -r \"https://cdn.oxfordnanoportal.com/software/analysis/dorado/dna_r10.4.1_e8.2_400bps_hac@v5.2.0.zip\" \n",
    "unzip /gxfs_work/geomar/smomw681/DORADO/dna_r10.4.1_e8.2_400bps_hac@v5.2.0.zip\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "993338e5",
   "metadata": {},
   "source": [
    "## Bioinformatic analysis process\n",
    "\n",
    "1. **Basecalling, demultiplexing** and **adapter trimming** using Dorado\n",
    "    - Dorado v.1.3.0\n",
    "    - Script\n",
    "        - Basecalling and demultiplexing: N16S_Run1_Dorado_BSC_Demux.sh, N16S_Run2_Dorado_BSC_Demux.sh\n",
    "        - Adapter Trimming: N16S_2_Trimming.sh\n",
    "\n",
    "2. **Renaming** the fastq files and the barcodes\n",
    "    - Script: N16S_2_1_ID_replacement.sh\n",
    "    - File location: /gxfs_work/geomar/smomw681/NANOPORE_DATA/DEMULTIPLEXED/NANOPORE_16S_RENAMED\n",
    "        - File name: $Sample_ID_renamed.fastq\n",
    "        - ID: $Sample_ID_(count)\n",
    "\n",
    "3. Data **filtering**: \n",
    "    - chopper v0.12.0\n",
    "    - Script: N16S_3_1_chopper.sh\n",
    "    - Kept 2838529 reads out of 3451719 reads ()\n",
    "    - Output: /gxfs_work/geomar/smomw681/NANOPORE_DATA/DEMULTIPLEXED/NANOPORE_16S_FILTERED\n",
    "    - All trimmed and filtered files should be **concatenated** in one file: \n",
    "        - Location: /gxfs_work/geomar/smomw681/NANOPORE_DATA/CONCATENATED/N16S_Seq_concat_filt.fastq\n",
    "\n",
    "4. Raw and filtered data **QC**: fastQC and multiQC\n",
    "    - fastQC 0.12.1\n",
    "    - multiQC 1.32\n",
    "    - Script: N16S_3_2_QC.sh\n",
    "    - Output location: \n",
    "        - Raw: /gxfs_work/geomar/smomw681/NANOPORE_DATA/QC_NANOPORE/QC_NANOPORE_16S_RAW\n",
    "        - Filtered: /gxfs_work/geomar/smomw681/NANOPORE_DATA/QC_NANOPORE/QC_NANOPORE_16S_FILTERED\n",
    "\n",
    "\n",
    "5. **16S rRNA identification**\n",
    "    - NanoCLUST\n",
    "        - Script:\n",
    "    - EMU:\n",
    "        - Script: N16S_ID/N16S_4_1_2_EMU.sh\n",
    "        - Output: \n",
    "            - Location:/gxfs_work/geomar/smomw681/NANOPORE_DATA/EMU/EMU_N16S_ind\n",
    "            - Files:\n",
    "                1. {sampleID}_renamed_filt_read-assignment-distributions.tsv\n",
    "                    - SeqID-TaxID matrix with read assignment probability\n",
    "                2. **{sampleID}_renamed_filt_rel-abundance.tsv**\n",
    "                    - Output of main interest\n",
    "                    - Relative abundance on species level in concatenated file (>0.01)\n",
    "                3. {sampleID}_renamed_filt_rel-abundance-threshold-0.0001.tsv\n",
    "                    - Relative abundance on species level in concatenated file (>0.0001)\n",
    "            - Concatenated into one file for the downstream analysis, visualization etc. \n",
    "                - Script: N16S_4_2_4_EMU_conca.sh\n",
    "                - Added sample ID to the last column \n",
    "                - File: N16S_EMU_rel_abundance_concat.tsv\n",
    "        - Visualization with R\n",
    "            - \n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "621422ac",
   "metadata": {},
   "source": [
    "## Sample specification and experiment information\n",
    "- Sample specification: Sample name (containing Sampling date, type, location and spot), short version \n",
    "    - without year (2025) and location as P1(SPZO1), P2(SPZO2), P3(POST)\n",
    "- Experiment information: barcode and batch \n",
    "    - Additional experiments with the DNA sample or corresponding RNA sample \n",
    "        - 16S: 16S rRNA V3/V4 sequencing on Illumina platform\n",
    "        - metaWGS: metagenomic whole genome sequencing on DNB-Seq platform\n",
    "        - metaRNA: metatranscriptomic sequencing on DNB-Seq platform \n",
    "\n",
    "- Format\n",
    "    Barcode.    short sample name   Other experiments\n",
    "\n",
    "###\n",
    "### Sequencing Run 1\n",
    "in N16S_Run1_barcode_sample_table.txt\n",
    "1. 0506_P1_FW_2_1   16S\n",
    "2. 0526_P1_FW_2_1   16S\n",
    "3. 0612_P1_FW_2_1   16S\n",
    "4. 0626_P1_FW_2_1   16S\n",
    "5. 0710_P1_FW_2_1   16S\n",
    "6. 0724_P1_FW_2_1   16S\n",
    "7. 0804_P1_FW_2_1   16S\n",
    "8. 0820_P1_FW_2_1   16S\n",
    "\n",
    "9. 0506_sp2_1   16S, metaWGS, metaRNA\n",
    "10. 0526_sp2_2   16S, metaWGS, metaRNA\n",
    "11. 0612_sp2_2   16S, metaWGS, metaRNA\n",
    "12. 0626_sp2_1   16S, metaWGS, metaRNA\n",
    "13. 0710_sp2_2   16S, metaWGS, metaRNA\n",
    "14. 0724_sp2_1   16S, metaWGS, metaRNA\n",
    "15. 0804_sp2_2_2   16S, metaWGS, metaRNA\n",
    "16. 0820_sp2_2   16S, metaWGS, metaRNA\n",
    "\n",
    "17. 0506_sp3_2   16S, metaWGS, metaRNA\n",
    "18. 0526_sp3_2   16S, metaWGS, metaRNA\n",
    "19. 0612_sp3_2   16S, metaWGS, metaRNA\n",
    "20. 0626_sp3_2   16S, metaWGS, metaRNA\n",
    "21. 0710_sp3_3_2   16S, metaWGS, metaRNA\n",
    "22. 0724_sp3_3   16S, metaWGS, metaRNA\n",
    "23. 0804_sp3_3   16S, metaWGS, metaRNA\n",
    "24. 0820_sp3_1   16S, metaWGS, metaRNA\n",
    "\n",
    "### Sequencing Run 2\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "214bbac9",
   "metadata": {},
   "source": [
    "## Samp"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "name": "python"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
