{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "653e3d3a",
   "metadata": {},
   "source": [
    "# Whole Genome Metagenomic Sequencing Data Analysis\n",
    "\n",
    "- 16 Freshwater sponge metaDNA samples extracted using AllPrep DNA/RNA kit\n",
    "- DNB-Seq by BGI Poland, at least 12 GB data per samples, with 95% of data over Q20\n",
    "- Data basecalled, adapter trimmed, low quality read removed by BGI \n",
    "- Reads were assembled to contig by David\n",
    "- \n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af3f4093",
   "metadata": {},
   "source": [
    "### Bioinformatic anaylsis steps\n",
    "\n",
    "0. QC: for cleaned reads (from BGI) and Error-corrected reads \n",
    "    - Script: DNB_1_1_QC.sh\n",
    "    - fastQC, multiQC for general quality metrics\n",
    "\n",
    "1. Assembly (by David)\n",
    "    - Scripts in folder: metaWGS_DNB_Seq/metaWGS_assem_David\n",
    "    - Quality improvement: PhiX174 removal, adapter trimming (trimmomatic), error-correction (bbmap tadpole.sh)\n",
    "    - Contig assembly (metaSPADes)\n",
    "\n",
    "2. Contig Assembly and gene prediction \n",
    "    1. File renaming\n",
    "        1. 1. Contig file renaming \n",
    "            - Script: DNB_2_1_ctg_rename.sh\n",
    "        1. 2. Error-corrected file renaming \n",
    "            - Script: DNB_2_1_EC_reads_rename.sh\n",
    "    2. Contig filtration by 500 bp length\n",
    "        - Script: DNB_2_2_FILTER_MIN_500BP.sh\n",
    "        - Perl script: SL_filter_contigs_on_size.pl\n",
    "    3. Contig ID renaming \n",
    "        - Script: DNB_2_3_RENAME_contigs.sh\n",
    "        - Perl script: SL_rename_fasta_id.pl\n",
    "\n",
    "    4. Contig Classification  \n",
    "        - Script: DNB_2_4_DeepMicroClass.sh\n",
    "        - DeepMicroClass (v1.0.3)\n",
    "        4. 1. Extraction of prokaryotic contigs\n",
    "            - Script: DNB_2_4_1_Extract_DMC_PROKS.sh\n",
    "\n",
    "    5. Prokaryotic gene prediction\n",
    "        - Script: DNB_2_5_Prodigal_gene_prediction.sh\n",
    "\n",
    "3. MAG Assembly\n",
    "    1. Mapping \n",
    "        1. 1. Mapping for coverage information \n",
    "            - Script: DNB_3_1_BamSamCov_before_MAG.sh\n",
    "                - BBmap (version: 39.42) for mapping and sorting\n",
    "                - metabat2 (version: 2.18) for coverage information\n",
    "            - Output files: .bam, .sorted.bam, coverage.txt \n",
    "                - File location: /gxfs_work/geomar/smomw681/BGI_META_WGS_DATA/BGI_DATA_SUNWOO/MAPPING\n",
    "        1. 2. Generate prokaryotic contigs fasta file mapped to error corrected reads\n",
    "            - Script: DNB_3_2_BBmap_proks_ctg_mapping.sh\n",
    "                - Bbmap\n",
    "            - Output: .EC_Proks_mapped.fq.gz\n",
    "        1. 3. QC for mapped prokaryotic contigs files\n",
    "            - \n",
    "    \n",
    "    2. Binning "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f9a567e4",
   "metadata": {
    "vscode": {
     "languageId": "plaintext"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "369ee967",
   "metadata": {
    "vscode": {
     "languageId": "plaintext"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "403f46a5",
   "metadata": {},
   "source": [
    "### Samples\n",
    "- Metagenomic freshwater sponge DNA samples from Sp2 and Sp3 from SPZO from May-August"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a9d7fed3",
   "metadata": {},
   "source": [
    "### Conda Environment Setup\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c39b1342",
   "metadata": {
    "vscode": {
     "languageId": "plaintext"
    }
   },
   "outputs": [],
   "source": [
    "# Assembly\n",
    "    conda create --name Assembly -y \n",
    "    conda activate Assembly    \n",
    "    conda install conda-forge::libgcc-ng=14.2.0     # for the current version of spades and bbmap\n",
    "    conda install bioconda::spades=4.0.0 -y   #or v4.1.0\n",
    "    conda install bioconda::trimmomatic=0.39 -y  \n",
    "    conda update trimmomatic # 0.40\n",
    "    conda install bioconda::bbmap # v39.52   # updated 08.01.26\n",
    "    conda install bioconda::samtools # 1.23     # updated 08.01.26\n",
    "    conda install conda-forge::pigz=2.8 -y\n",
    "    conda install bioconda::filtlong #0.3.1\n",
    "\n",
    "# MAG prep and after\n",
    "    conda create -n MAG\n",
    "    conda activate MAG\n",
    "    #conda install bioconda::deepmicroclass=1.0.3 -y \n",
    "    conda install python=3.12.9 -y\n",
    "    pip install DeepMicroClass # without pytorch: use CPU only\n",
    "    pip install requests #v2.32.3\n",
    "    conda install bioconda::checkm-genome -y #=1.2.3\n",
    "    conda install bioconda::prodigal -y #=2.6.3\n",
    "    conda install bioconda::pprodigal -y #=1.0.1\n",
    "    conda install bioconda::gtdbtk -y #=2.4.0\n",
    "    # db for gtdbtk also downloaded: run script 3_0 \n",
    "    conda install bioconda/label/cf201901::mash # v2.3\n",
    "\n",
    "# MAG_construction and check\n",
    "    conda create -n METABAT2\n",
    "    conda activate METABAT2\n",
    "    conda install bioconda::metabat2=2.18   # updated 08.01.26\n",
    "    # metabat2 must be installed anew according to the description on the official github site due to negative coverage problem\n",
    "    # https://bitbucket.org/berkeleylab/metabat/src/master/INSTALL.md\n",
    "    module load gcc12-env/12.3.0\n",
    "    module load gcc/12.3.0\n",
    "    module load boost/1.83.0\n",
    "    module load cmake/3.27.4\n",
    "    conda install bioconda::checkm2=1.1.0\n",
    "    # checkm2 database in /gxfs_work/geomar/smomw681/.conda/envs/METABAT2/checkm_data/\n",
    "    # or download it to a custom directory and set as environmental variable\n",
    "        checkm2 database --download --path  /gxfs_work/geomar/smomw681/DATABASES/CheckM_db/\n",
    "        export CHECKM2DB=\"/gxfs_work/geomar/smomw681/DATABASES/CheckM_db/CheckM2_database\"\n",
    "        conda env config vars set CHECKM2DB=\"/gxfs_work/geomar/smomw681/DATABASES/CheckM_db/CheckM2_database\"\n",
    "    conda install bioconda::coverm=0.7.0\n",
    "    conda install bioconda::drep # \n",
    "    # install dependencies of drep\n",
    "    conda install bioconda/label/cf201901::mash # v2.1\n",
    "    module load boost/1.83.0\n",
    "    module load gcc/12.3.0\n",
    "    module load gsl/2.7.1\n",
    "    conda install bioconda::centrifuge -y # v1.0.4.2\n",
    "    # conda install bioconda/label/cf201901::fastani -y # v1.1\n",
    "        # But couldn't solve the dependency problem of fastANI. conflicting with gsllib\n",
    "        # made a separate environment because of dependency problem"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "name": "python"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
